### ISSUE TITLE
Soundcloud support

### ISSUE THREAD

**Issue by xijo** [avantrec/soco-cli#74]
Hey folks,

First of all thanks for the work you've put into this library.

I was wondering if it was possible to add soundcloud support to it. Afaics it should not be substantially different from how spotify works, right? 

Any idea how to start or how difficult that would be to add?

Thanks and greetings,
Johannes

**Comment by pwt** [comment]
Hi:

Thanks for your comments and sorry for the late response.

Are you referring to **ShareLink** support for SoundCloud, as currently provided for Apple, Spotify, Tidal and Deezer? I ask because we don't have general support for music services (because they're not robustly supported by the SoCo library).

Cheers,
Peter

**Comment by xijo** [comment]
Yes, exactly, I was asking about sharelink support for SoundCloud. I’d be
happy to work on a PR if you could provide some general direction on how to
start 😏

**Comment by pwt** [comment]
> Yes, exactly, I was asking about sharelink support for SoundCloud. I’d be happy to work on a PR if you could provide some general direction on how to start 😏

Thanks. SoundCloud ShareLink support would need to be added in the underlying [SoCo](https://github.com/SoCo/SoCo) library, within the ShareLink plugin, then SoCo-CLI would automatically acquire the capability.

I'll take a look to see how feasible this is.

**Comment by maqbeq2** [comment]
I have worked around this limitation using yt-dlp. Consider that I don't have a SC subscription so I am limited to SC's 128kpbs audio stream, using the following might suit you @xijo 
`yt-dlp -f http_mp3_0_0 -g [insert-soundcloud-link-here] | xargs sonos [your-speaker-name-here] play_uri`

**Comment by xijo** [comment]
Wow that is very smart, I‘ll give it a try! Thank you

**Comment by pwt** [comment]
> I have worked around this limitation using yt-dlp

That's very nice -- thanks.

(I did investigate adding SoundCloud to the SoCo library some time ago, but didn't get very far.)
